### PR TITLE
Add regression test for relative imports in tests under `lib`

### PR DESCRIPTION
**Pull request by jakemac53**
Validates the issue listed in https://github.com/dart-lang/test/issues/2142.

Skipped for now since it fails in all configurations.

### CODE REVIEW

File: integration_tests/regression/test/issue_2142_test.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
+@Skip('https://github.com/dart-lang/test/issues/2142')
+library;
+
+import 'package:test/test.dart';
+
+// ignore: avoid_relative_lib_imports
+import '../lib/issue_2142/import.dart';
```
Comment (jakemac53) [inline integration_tests/regression/test/issue_2142_test.dart:7]: Note that this one probably would be broken either way?

Comment (natebosch) [reply integration_tests/regression/test/issue_2142_test.dart:7]: Yes, I think we should not attempt to support `../lib`. I'd be happy to check for this and block it if it's easy.

**Comment by natebosch** [comment]
Oops, I meant to push a new branch and open a PR to pull into this branch, I hadn't intended to push on this branch. Sorry @jakemac53 !

**Comment by jakemac53** [comment]
> Oops, I meant to push a new branch and open a PR to pull into this branch, I hadn't intended to push on this branch. Sorry @jakemac53 !

No worries, should I revert or want to land that together with the test?

**Comment by natebosch** [comment]
> No worries, should I revert or want to land that together with the test?

 I reverted it on this branch. I think we should go ahead and land the test and then handle the fix in a separate PR.

File: integration_tests/regression/lib/issue_2142/test.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@Skip('https://github.com/dart-lang/test/issues/2142')
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply integration_tests/regression/lib/issue_2142/test.dart:1]: Presumably (but I have no idea). I will add one.

**Review by natebosch — APPROVED** [review]

**Pull request merged**